SPECIFICATION.md

```diff
@@ -122,19 +122,19 @@ If we had received the hash for tree index 5 before and verified it then we woul
 
 It should be noted that this allows us to have random access to any block in the feed with content verification in a single round trip
 
-As an optimization we can use the remote's "have" messages (see the "Wire Protocol" section) to figure out which hashes it already has. For example if the remote has block #3 then we do not need to send any hashes since it will already have verified hash 6 and 5.
+As an optimization we can use the remote's "have" messages (see the ["Wire Protocol" section](#messageshave)) to figure out which hashes it already has. For example if the remote has block #3 then we do not need to send any hashes since it will already have verified hash 6 and 5.
 
 Hyperdrive's content integrity approach is similar to parts of [ppspp](https://tools.ietf.org/html/rfc7574) and additional information can found in that specification.
 
 ## Deduplication
 
-Assuming you have two different feeds that are sharing similar data we want to able to fetch as little as possible duplicate data. The merkle tree structure helps us achieve this by content addressing the feeds.
+Assuming you have two different feeds that are sharing similar data, we want to minimize fetching of duplicate data. The merkle tree structure helps us achieve this by content addressing the feeds.
 
-For example if we were to produce the exact same feed on two different computers using the technique described in the above section the feeds would end up with the same root tree hashes and the same feed id and it would therefore be the same feed.
+For example if we were to produce the exact same feed on two different computers using the technique described in the above section, the feeds would end up with the same root tree hashes and the same feed id. The two feeds would therefore be the same feed.
```

Comment:
Should we add an anchor link to "described in the above section" to the corresponding section?  (Man I wish MD had a few RST features like section references)